pyvista/core/dataset.py

```diff
@@ -2523,3 +2523,54 @@ def cell_type(self, ind: int) -> int:
 
         """
         return self.GetCellType(ind)
+
+    def point_is_inside_cell(
+        self, ind: int, point: Union[VectorArray, NumericArray]
+    ) -> Union[int, np.ndarray]:
+        """Return whether a point is inside a cell.
+
+        Parameters
+        ----------
+        ind : int
+            Cell ID.
+
+        point : Sequence(float) or np.ndarray
+            Coordinates of point to query (length 3) or a ``numpy`` array of ``n``
+            points with shape ``(n, 3)``.
+
+        Returns
+        -------
+        bool or numpy.ndarray
+            Whether point(s) is/are inside cell.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> mesh = examples.load_hexbeam()
+        >>> mesh.cell_bounds(0)
+        (0.0, 0.5, 0.0, 0.5, 0.0, 0.5)
+        >>> mesh.point_is_inside_cell(0, [0.2, 0.2, 0.2])
+        True
+
+        """
+        if not 0 <= ind < self.n_cells:
+            raise ValueError(f"ind must be >= 0 and < {self.n_cells}, got {ind}")
```

Comment:
Should we add a `TypeError` for non-integer `ind`? I don't insist on doing that since most weird cases will fail loudly with this check. Only the "passed in a float by accident" case might be relevant for this, assuming that such accidental floats are not int-valued floats.